sdk/iOS/test/MSSyncTableTests.m

```diff
@@ -1082,9 +1083,348 @@ -(void) testPushNetworkTimeoutWithErroredOps
+// - Returns the number of items pushed to the server.
+-(int) performPushWithCancellationPoint:(int) requestedCancellationPoint serverResponseCode:(NSNumber *)serverResponseCode
+{
+    // Initialization
+    MSMultiRequestTestFilter *filter = [MSMultiRequestTestFilter testFilterWithStatusCodes:@[serverResponseCode] data:@[@"{\"id\": \"id1\", \"text\":\"server\"}"] appendEmptyRequest:YES];
+    MSClient *filteredClient = [client clientWithFilter:filter];
+    MSSyncTable *todoTable = [filteredClient syncTableWithName:TodoTableNoVersion];
+
+    // Start with a clean slate
+    [[todoTable forcePurgeWithCompletion:^(NSError *error) {
+        XCTAssertNil(error, @"error should have been nil.");
+    }] waitUntilFinished];
+
+    __block NSOperation *pushOperation;
+
+    // Define filters
+    ((MSTestFilter *)filter.testFilters[0]).onInspectRequest = ^(NSURLRequest *request) {
+
+        // Cancellation point 0
+        if (requestedCancellationPoint == 0)
+            [pushOperation cancel];
+        
+        return request;
+    };
+    
+    ((MSTestFilter *)filter.testFilters[0]).onInspectResponseData = ^(NSURLRequest *request, NSData *data) {
+
+        // Cancellation point 1
+        if (requestedCancellationPoint == 1)
+            [pushOperation cancel];
+        
+        return data;
+    };
+    
+    // Perform an insert
+    XCTestExpectation *insertExpectation = [self expectationWithDescription:@"insert expectation"];
+    [todoTable insert:@{ @"id" : @"id1", @"text" : @"client" } completion:^(NSDictionary *item, NSError *error) {
+        [insertExpectation fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+    
+    // Perform a push
+    pushOperation = [filteredClient.syncContext pushWithCompletion:^(NSError *error) {
+        
+        // Cancellation point 2
+        if (requestedCancellationPoint == 2)
+            [pushOperation cancel];
+    }];
+    
+    // Cancellation point 3
+    if (requestedCancellationPoint == 3)
+        [pushOperation cancel];
+
+    [self verifyFinished:pushOperation];
+
+    int synchronizedItemCount = [self synchronizedItemCount];
+
+    // Perform another insert
+    insertExpectation = [self expectationWithDescription:@"insert expectation"];
+    [todoTable insert:@{ @"id" : @"id2", @"text" : @"client" } completion:^(NSDictionary *item, NSError *error) {
+        [insertExpectation fulfill];
+    }];
+    
+    [self waitForExpectationsWithTimeout:10 handler:nil];
+
+    XCTAssertEqual([self performVanillaPush], 2, "Expected all inserted items to have been pushed to the server by now");
+
+    return synchronizedItemCount;
+}
+
+// Performs a push operation and returns the count of all client items pushed so far
+-(int) performVanillaPush
+{
+    // Define a filter that returns an appropriate server response for the requested item
+    MSTestFilter *filter = [MSTestFilter new];
+    MSClient *filteredClient = [client clientWithFilter:filter];
```

Comment:
Yes, that's intentional. This push is performed without cancellation after a prior push operation was cancelled. The idea is to see if this completes successfully to verify that the cancelled push operation did not mess up internal data structures such that no future push operations go through.